clearwater-cluster-manager.root/usr/share/clearwater/clearwater-cluster-manager/scripts/mark_node_failed.py

```diff
@@ -9,13 +9,14 @@
 from os import sys
 import etcd
 import logging
+from metaswitch.clearwater.cluster_manager.cluster_state import \
+    ClusterInfo
 from metaswitch.clearwater.cluster_manager.etcd_synchronizer import \
     EtcdSynchronizer
 from metaswitch.clearwater.cluster_manager.null_plugin import \
     NullPlugin
-from metaswitch.clearwater.etcd_shared.plugin_utils import run_command
 
-def make_key(site, node_type, datatore, etcd_key):
+def make_key(site, node_type, datastore, etcd_key):
```

Comment:
I reckon python lets functions just inherit the outer stack whenever they are called. So, when `make_key` was called, the `datastore` on line 34 was somehow in scope (and `datatore` is totally ignored).

I really hope that's not true though - but seems like the only explanation...